src/intrinsics.cpp

```diff
@@ -509,6 +509,11 @@ static jl_cgval_t generic_bitcast(jl_codectx_t &ctx, const jl_cgval_t *argv)
     bool isboxed;
     Type *vxt = julia_type_to_llvm(ctx, v.typ, &isboxed);
 
+    if (v.typ == jl_bottom_type) {
+        emit_error(ctx, "bitcast: expected primitive type value for second argument");
```

Comment:
you don't need to emit anything in unreachable code blocks
```suggestion
```

Reply:
Oh right, bottom values are impossible. I'll update the test too.

Reply:
You also might want to move this into the caller, where we compute the `argv`, since any of the other intrinsics here might have this issue too (the probably don't, since usually they need more exact type info, but might be good anyways)

Reply:
OK, pushed that change.